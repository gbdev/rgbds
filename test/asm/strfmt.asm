DEF VAL EQUS STRFMT("Hello %s! I am %d years old today!", "world", $f)
PRINTLN "{VAL}"

DEF N = -42
PRINTLN STRFMT("signed %010d == unsigned %010u", N, N)

DEF N = 112
DEF FMT EQUS "X"
PRINTLN STRFMT("\tdb %#03{s:FMT} %% 26\t; %#03{FMT}", N, N % 26)

PRINTLN STRFMT("%d = %#x = %#b = %#o != %f", 42, 42, 42, 42, 42.0)

DEF TEMPLATE EQUS "\"%s are %s\\n\""
PRINT STRFMT(TEMPLATE, "roses", "red")
PRINT STRFMT(TEMPLATE, "violets", "blue")
PRINT STRFMT(TEMPLATE, "void", 0, "extra")

PRINTLN STRCAT(STRFMT(STRFMT("%%%s.%d%s", "", 9, "f"), 3.14159), \
	STRFMT(" ~ %s", STRFMT("%s%x", "thr", 238)))

DEF N = 1.23456
PRINTLN STRFMT("%.f -> %.3f -> %f", N, N, N)

PRINTLN STRFMT("%d eol %", 1)

PRINTLN STRFMT("invalid %w spec", 42)

PRINTLN STRFMT("one=%d two=%d three=%d", 1)
