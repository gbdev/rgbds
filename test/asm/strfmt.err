error: strfmt.asm(16):
    Formatting number as type 's'
error: strfmt.asm(16):
    STRFMT: 1 unformatted argument(s)
error: strfmt.asm(24):
    STRFMT: Illegal '%' at end of format string
error: strfmt.asm(26):
    STRFMT: Invalid format spec for argument 1
error: strfmt.asm(28):
    STRFMT: Not enough arguments for format spec, got: 1, need: 3
error: Assembly aborted (5 errors)!
