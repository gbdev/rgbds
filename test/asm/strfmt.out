Hello world! I am 15 years old today!
signed -000000042 == unsigned 4294967254
	db $70 % 26	; $08
42 = $2a = %101010 = &52 != 42.00000
roses are red
violets are blue
void are 0
3.141586304 ~ three
1 -> 1.235 -> 1.23456
1 eol %
invalid % spec
one=1 two=% three=%
